MongoDB ODM 2.3 blog post

IonBazan opened the pull request.

malarzm commented on source/blog/2021-12-01-mongodb-odm-2.3.md:
```diff
@@ -0,0 +1,126 @@
+---
+title: "New Release: Doctrine MongoDB ODM 2.3 with Attributes, JSON Schema Validation and more"
```
```suggestion
title: "New Release: Doctrine MongoDB ODM 2.3 with Attributes, JSON Schema Validation, and more"
```

malarzm commented on source/blog/2021-12-01-mongodb-odm-2.3.md:
```diff
@@ -0,0 +1,126 @@
+## JSON Schema Validation
+
+MongoDB ≥ 3.6 offers the capability to validate documents during
+insertions and updates through a JSON schema associated to the collection. 
```
```suggestion
insertions and updates through a JSON schema associated with the collection. 
```
but not sure about this one

malarzm commented on source/blog/2021-12-01-mongodb-odm-2.3.md:
```diff
@@ -0,0 +1,126 @@
+[See MongoDB documentation](https://docs.mongodb.com/manual/core/schema-validation/).
+
+Doctrine MongoDB ODM now provides a way to take advantage of this functionality
+thanks to the new `#[Validation]` mapping.
```
Given our "annotation reference" page is still not showing up in the menu I'd link to https://www.doctrine-project.org/projects/doctrine-mongodb-odm/en/latest/reference/annotations-reference.html#validation directly

malarzm commented on source/blog/2021-12-01-mongodb-odm-2.3.md:
```diff
@@ -0,0 +1,126 @@
+
+## Psalmified APIs
+
+Improved the documentation to make sure static analysis tools and IDEs know
```
```suggestion
In-code documentation has been immensely improved to make sure static analysis tools and IDEs know
```

malarzm commented on source/blog/2021-12-01-mongodb-odm-2.3.md:
```diff
@@ -0,0 +1,126 @@
+
+Improved the documentation to make sure static analysis tools and IDEs know
+about the right document classes returned from `DocumentManager`,
+`DocumentManager`, `ClassMetadata` and other public ODM APIs. This includes generics support
```
```suggestion
`ClassMetadata`, and other public APIs. This includes generics support
```
- DocumentManager was listed twice
- Oxford comma

malarzm commented on source/blog/2021-12-01-mongodb-odm-2.3.md:
```diff
@@ -0,0 +1,126 @@
+Improved the documentation to make sure static analysis tools and IDEs know
+about the right document classes returned from `DocumentManager`,
+`DocumentManager`, `ClassMetadata` and other public ODM APIs. This includes generics support
+when you extend `DocumentRepository`.
```
```suggestion
for your own repositories extending `DocumentRepository`.
```

malarzm commented on source/blog/2021-12-01-mongodb-odm-2.3.md:
```diff
@@ -0,0 +1,126 @@
+- Annotation classes no longer extend `Doctrine\Common\Annotations\Annotation` class
+- Annotation arguments switched to `@NamedArgumentConstructor` for Attribute compatibility
+- `@Inheritance` annotation has been removed as it was never used
+- Document Namespace Aliases (`'MyBundle:User`) - use fully qualified class names instead (`User::class`)
```
I'd add a link to https://github.com/doctrine/mongodb-odm/blob/2.3.x/UPGRADE-2.3.md

IonBazan commented:
@malarzm thanks for your review. I've fixed the issues you mentioned and updated the post date to 4 Dec. Feel free to change that according to the actual publish date.
